mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala

```diff
@@ -0,0 +1,740 @@
+   * by Blei et al., but are called "phi" in many later papers such as Asuncion et al., 2009.
+   *
+   * If set to -1, then topicConcentration is set automatically.
+   *  (default = -1 = automatic)
+   *
+   * Optimizer-specific parameter settings:
+   *  - EM
+   *     - Value should be > 1.0
+   *     - default = 0.1 + 1, where 0.1 gives a small amount of smoothing and +1 follows
+   *       Asuncion et al. (2009), who recommend a +1 adjustment for EM.
+   *  - Online
+   *     - Value should be >= 0
+   *     - default = (1.0 / k), following the implementation from
+   *       [[https://github.com/Blei-Lab/onlineldavb]].
+   * @group param
+   */
+  @Since("1.6.0")
+  final val topicConcentration = new DoubleParam(this, "topicConcentration",
+    "Concentration parameter (commonly named \"beta\" or \"eta\") for the prior placed on topic'" +
+      " distributions over terms.", (beta: Double) => beta == -1 || beta >= 0.0)
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getTopicConcentration: Double = $(topicConcentration)
+
+  /**
+   * Alias for [[getTopicConcentration]]
+   * @group getParam
+   */
+  @Since("1.6.0")
+  def getBeta: Double = getTopicConcentration
+
+  /**
+   * Optimizer or inference algorithm used to estimate the LDA model, specified as a
+   * [[LDAOptimizer]] type.
+   * Currently supported:
+   *  - Online Variational Bayes: [[OnlineLDAOptimizer]] (default)
+   *  - Expectation-Maximization (EM): [[EMLDAOptimizer]]
+   * @group param
+   */
+  @Since("1.6.0")
+  final val optimizer = new Param[LDAOptimizer](this, "optimizer", "Optimizer or inference" +
+    " algorithm used to estimate the LDA model")
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getOptimizer: LDAOptimizer = $(optimizer)
+
+  // Developers should override these setOptimizer() methods.  These are defined here to
+  // ensure identical behavior when setting the optimizer using a String.
+  /** @group setParam */
+  @Since("1.6.0")
+  def setOptimizer(value: LDAOptimizer): this.type = set(optimizer, value)
+
+  /**
+   * Set [[optimizer]] by name (case-insensitive):
+   *  - "online" = [[OnlineLDAOptimizer]]
+   *  - "em" = [[EMLDAOptimizer]]
+   * @group setParam
+   */
+  @Since("1.6.0")
+  def setOptimizer(value: String): this.type = value.toLowerCase match {
+    case "online" => setOptimizer(new OnlineLDAOptimizer)
+    case "em" => setOptimizer(new EMLDAOptimizer)
+    case _ => throw new IllegalArgumentException(
+      s"LDA was given unknown optimizer '$value'.  Supported values: em, online")
+  }
+
+  /**
+   * Output column with estimates of the topic mixture distribution for each document (often called
+   * "theta" in the literature).  Returns a vector of zeros for an empty document.
+   *
+   * This uses a variational approximation following Hoffman et al. (2010), where the approximate
+   * distribution is called "gamma."  Technically, this method returns this approximation "gamma"
```

Comment:
nit: `gamma` is an implementation detail, we could just say that this returns the variational approximation to the true topic dirichlet posterior